### PR TITLE
[Bug] Fix some issues with async file downloads

### LINKED ISSUE #11229: 🐛 [UAT] Candidate CSV and Word downloads fail for some candidates
## 🐛 Bug

On UAT, file downloads (CSV and Word docs) are working as intended for some candidates, but not for others.  https://github.com/GCTC-NTGC/gc-digital-talent/releases/tag/v2.28.0

## 🦋 Expected Behaviour

CSV and Word downloads should work regardless of the candidate or candidates selected.

## 🕵️ Details

For some candidates, if you try to download as a CSV, you never get a file-download notification. If you select multiple candidates, it seems like any group which contains one of these poison candidates also fails.

For some candidates, if you try to download profile as a word doc, you get the notification and can download the file, but Word crashes when trying to open it. If you select multiple candidates, it seems like any group which contains one of these poison candidates also fails in the same way.

Interestingly, one candidate might work for the CSV but not for the Word doc.

## 📋 Steps to Reproduce

Reproducing on UAT
1. Log in as a Platform Admin / Request Responder
2. Go to https://uat-talentcloud.tbs-sct.gc.ca/en/admin/pools/122160cd-008d-4693-890e-6e05657a7f41/pool-candidates
3. Try to download candidate 72d70f2a-f701-4060-ab96-affc2c847185 (Rosalie). This should work as a CSV download and a Profile download.
4. Try to download candidate 6d093e10-f109-4b10-9fbb-228dc6ebb8d2 (Peter). This should work as a Profile download but not as a CSV.
5. Try to download candidate 9a3aac90-3671-4494-ac92-843b8a0a7599 (Jean). This should work for neither download.
6. Check multiple rows and explore downloading different combinations of candidates.

## 📸 Screenshot

Add a screenshot (if possible).

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the bug was addressed.

- [ ] All candidates can be downloaded as CSV (including on UAT)
- [ ] All candidates, when downloaded as word docs, can be opened successfully in Word.

### PR DESCRIPTION
**Pull request by petertgiles**
🤖 Resolves #11229 

## 👋 Introduction

This branch corrects a few issues with the async file downloads.

## 🕵️ Details

Issue: No error reported when something goes wrong with generation
Fix: Switch "catch exception" to "catch throwable"

Issue: A null education requirement causes CSV crash
Fix: Switch the method signature to accept a nullable parameter

Issue: An ampersand causes DOCX corruption
Fix: Set PHPOffice to escape illegal XML characters

### Issue 1 - No error message

1. Check out the first commit of this branch: 235b8f3231e299c4f8ea1f9d8ee887c5e2f34420 and rebuild
2. Manually set the `education_requirement_option` of a `pool_candidates` row to null in the database
3. Start the queue worker
4. Log into the app as a admin and try the "Download CSV" command for the edited pool candidate
5. Observe the error notification

### Issue 2 - CSV crash

1. Check out the head of this branch
2. Manually set the `education_requirement_option` of a `pool_candidates` row to null in the database
3. Start the queue worker
4. Log into the app as a admin and try the "Download CSV" command for the edited pool candidate
5. Observe that the file is successfully generated with an empty education requirement column

### Issue 3 - DOCX corrupted

1. Check out the head of this branch
2. Log in as a user and add an "&", for example, to an experience attached to a pool candidate
3. Start the queue worker
4. Log into the app as a admin and try the "Download profile" command for the edited pool candidate
5. Observe that the file can be opened in Word with the "&" visible

### CODE REVIEW

File: api/app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -24,6 +24,10 @@ public function boot()
         // https://laravel.com/docs/9.x/eloquent#configuring-eloquent-strictness
         Model::shouldBeStrict(! $this->app->isProduction());
 
+        // https://github.com/PHPOffice/PHPWord/issues/2524#issuecomment-1847981808
+        // https://phpoffice.github.io/PHPWord/usage/introduction.html#output-escaping
+        \PhpOffice\PhpWord\Settings::setOutputEscapingEnabled(true);
```
Comment (esizer) [inline api/app/Providers/AppServiceProvider.php:29]: Odd that this is not the default :face_with_spiral_eyes:

Comment (petertgiles) [reply api/app/Providers/AppServiceProvider.php:29]: Yeah, for sure.  The explanation on the website is that they want to maintain backward comparability.  Just increment the major version number and fix it already!

Comment (petertgiles) [reply api/app/Providers/AppServiceProvider.php:29]: Reading through that page, we might want to update the paper to letter, as well.

https://phpoffice.github.io/PHPWord/usage/introduction.html#default-paper

Comment (esizer) [reply api/app/Providers/AppServiceProvider.php:29]: Must be an EU developed library :stuck_out_tongue:

**Review by tristan-orourke — APPROVED** [review]
Looks good to me!

**Pull request merged**